src/EventStore.Core/Services/Transport/Http/Controllers/AdminController.cs

```diff
@@ -41,16 +44,86 @@ private void OnPostShutdown(HttpEntityManager entity, UriTemplateMatch match)
 
         private void OnPostScavenge(HttpEntityManager entity, UriTemplateMatch match)
         {
-            if (entity.User != null && (entity.User.IsInRole(SystemRoles.Admins) || entity.User.IsInRole(SystemRoles.Operations)))
+            int startFromChunk = 0;
+
+            var startFromChunkVariable = match.BoundVariables["startFromChunk"];
+            if (startFromChunkVariable != null)
             {
-                Log.Info("Request scavenging because /admin/scavenge request has been received.");
-                Publish(new ClientMessage.ScavengeDatabase(new NoopEnvelope(), Guid.Empty, entity.User));
-                entity.ReplyStatus(HttpStatusCode.OK, "OK", LogReplyError);
-            }
-            else
+                if (!int.TryParse(startFromChunkVariable, out startFromChunk) || startFromChunk < 0)
+                {
+                    SendBadRequest(entity, "startFromChunk must be a positive integer");
+                    return;
+                }
+            } 
+            
+            int threads = 1;
+
+            var threadsVariable = match.BoundVariables["threads"];
```

Comment:
Should be `QueryParameters` instead of `BoundVariables`

Reply:
Fixed with addition of threads to the template.